lib/token.cpp

```diff
@@ -2371,6 +2371,15 @@ bool Token::hasKnownIntValue() const
     });
 }
 
+bool Token::hasKnownBoolValue() const
+{
+    if (!mImpl->mValues)
+        return false;
+    return std::any_of(mImpl->mValues->begin(), mImpl->mValues->end(), [](const ValueFlow::Value& value) {
+        return value.isIntValue() && (value.isKnown() || (value.intvalue == 0 && value.isImpossible()));
```

Comment:
`valueFlowInferCondition` should make these values known, instead of having the checkers deal with this logic.